src/jaegertracing/samplers/ProbabilisticSampler.h

```diff
@@ -53,6 +52,20 @@ class ProbabilisticSampler : public Sampler {
     double _samplingRate;
     uint64_t _samplingBoundary;
     std::vector<Tag> _tags;
+
+    static uint64_t computeSamplingBoundary(long double samplingRate)
+    {
+        const long double maxRandNumber = kMaxRandomNumber;
```

Comment:
Clarification - the trace ID is used for sampling in the Go client merely as a performance optimization. There's no functional dependency on doing it that way. In the Node.js client we do use a new random number for the sampling decision since Node cannot do uint64 arithmetics.

Reply:
Interesting. Makes sense. That is okay with me.

Reply:
@isaachier This does not require `long double` to be 80-bit to work.

All you need to handle the overflow is a conditional like [this](https://github.com/rnburn/cpp-client/blob/overflow/src/jaegertracing/samplers/ProbabilisticSampler.h#L63).

`long double` will still give you a better answer if extended precision is available, but it will still work if `long double` == `double`.

Reply:
Shouldn't it be `>=`. And you wrote yourself that using a standard random number generator would make more sense here. Do you no longer feel that way?

Reply:
Since `samplingRate <= 1.0`, `samplingRate * maxRandNumber` is always less than or equal to `maxRandNumber`; so it's not necessary.

Using `std::uniform_real_distribution` would be straightforward, but I don't see anything wrong with putting in a small fix to immediately address the overflow issue.

Reply:
OK got it. Overall, I hate dealing with overflow, but the random number generator involves changing a number of unit tests as well. So I am happy to accept this, but if you can (again), please merge from master and push so we can see coverage info.

Reply:
Merged master.